test/functional/p2p_relay_transactions.py

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env python3
+# Copyright (c) 2017-2021 The Bitcoin Core developers
+# Copyright (c) 2022 RBB S.r.l
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or http://www.opensource.org/licenses/mit-license.php.
+
+from test_framework import mintlayer_hash
+from test_framework.util import (assert_raises_rpc_error)
+from test_framework.test_framework import BitcoinTestFramework
+import scalecodec
+import time
+
+class RelayTransactions(BitcoinTestFramework):
+    def set_test_params(self):
+        self.num_nodes = 2
+
+    def setup_network(self):
+        self.setup_nodes()
+        self.connect_nodes(0, 1)
+        self.sync_all(self.nodes[0:2])
+
+    def assert_mempool_contains_tx(self, n, tx_id):
+        for _ in range(5):
+            if self.nodes[n].mempool_contains_tx(tx_id):
+                break
+            time.sleep(1)
+        assert self.nodes[n].mempool_contains_tx(tx_id)
+
+    def run_test(self):
+        signed_tx_obj = scalecodec.base.RuntimeConfiguration().create_scale_object('SignedTransaction')
+        base_tx_obj = scalecodec.base.RuntimeConfiguration().create_scale_object('TransactionV1')
+
+        # Try to submit an invalid transaction.
+        tx = {
+            'version': 1,
+            'flags': 0,
+            'inputs': [],
+            'outputs': [],
+            'time_lock': 0,
+        }
+        signed_tx = {
+            'transaction': tx,
+            'signatures': [],
+        }
+        encoded_tx = signed_tx_obj.encode(signed_tx).to_hex()[2:]
+        tx_id = scalecodec.ScaleBytes(mintlayer_hash(base_tx_obj.encode(tx).data)).to_hex()[2:]
```

Comment:
We should create some helper methods to encode stuff into hex and get hex ID of an object. This is a common task and is already duplicated in the mempool test.

Reply:
Agreed. Someone please open the issue and explain the problem in better details. I didn't do this often so I can't describe it properly.

Reply:
https://github.com/mintlayer/mintlayer-core/issues/849